arangod/Agency/Agent.cpp

```diff
@@ -451,14 +488,13 @@ void Agent::sendAppendEntriesRPC() {
       // Note that despite compaction this vector can never be empty, since
       // any compaction keeps at least one active log entry!
 
-      if (unconfirmed.empty()) {
+      if (unconfirmed.empty() && !raiseLastConfirmed) {
         LOG_TOPIC(ERR, Logger::AGENCY) << "Unexpected empty unconfirmed: "
           << "lastConfirmed=" << lastConfirmed << " commitIndex="
           << commitIndex;
+        TRI_ASSERT(false);
       }
 
-      TRI_ASSERT(!unconfirmed.empty());
-
       if (unconfirmed.size() == 1) {
```

Comment:
Is this still correct with `raiseLastConfirmed`? I think if `raiseLastConfirmed` is `true`, we should not handle the size 1 case in this way.

Reply:
fixed. manually testing